### PR TITLE
Create subscription list

### PR DESCRIPTION
**Pull request by sarvesh371**
Built two apis
- Create subscription list
- Delete subscription list

### CODE REVIEW

File: backend/controllers/subscription.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { Request, Response } from "express";
+import { runQuery } from "../utils/postgres";
+
+export async function CreateSubscriptionList(req: Request, res: Response): Promise<void> {
+    const { lister_wallet, name, details, amount, currency, period } = req.body;
+
+    try {
+        // Validate that we don't have same name subscription in the list
```
Comment (Codimow) [inline backend/controllers/subscription.ts:8]: const subscriptionList = await runQuery(
            `SELECT * FROM public.subscriptions_list WHERE LOWER(name) = LOWER($1)`,
            [name]
        );

File: backend/controllers/subscription.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            res.status(400).json({
+                status: "error",
+                data: [],
+                message: "There is already a subscription with this name !!",
+            });
+            return;
+        }
+        // Write the data to the database
+        await runQuery(
+            `INSERT INTO subscriptions_list
```
Comment (Codimow) [inline backend/controllers/subscription.ts:24]: await runQuery(
            `INSERT INTO subscriptions_list 
            (lister_wallet, name, details, amount, status, currency, period) 
            VALUES ($1, $2, $3, $4, 'ACTIVE', $5, $6)`,
            [lister_wallet, name, details, amount, currency, period]
        );

File: backend/controllers/subscription.ts (hunk 1 above)
Comment (sarvesh371) [reply backend/controllers/subscription.ts:8]: Updated

File: backend/config/index.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import * as dotenv from "dotenv";
```
Comment (kiritocode1) [inline backend/config/index.ts:1]: you can replace this with just 
```typescript 
import "dotenv/config"

File: backend/config/index.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+if (missingEnvVars.length > 0) {
+  throw new Error(
+    `Missing or empty environment variables: ${missingEnvVars.join(", ")}`
+  );
+}
+
+const { DB_HOST, DB_USER, DB_DATABASE, DB_PASSWORD, DB_PORT, SERVER_PORT, SSL } =
+  process.env;
+
+export { DB_HOST, DB_USER, DB_DATABASE, DB_PASSWORD, DB_PORT, SERVER_PORT, SSL };
```
Comment (kiritocode1) [inline backend/config/index.ts:28]: no , please don't do so unless you explicitly say so , best approach would be to extend the node interface. an even  better approach would be to use something called env.t3.gg for these things

Comment (kiritocode1) [reply backend/config/index.ts:28]: the reason for not doing this is  there is no guarentee that this script would be run unless mentioned so

Comment (kiritocode1) [reply backend/config/index.ts:28]: please refer to this code 
```typescript
import { z } from "zod";
 
const envVariables = z.object({
  DATABASE_URL: z.string(),
  CUSTOM_STUFF: z.string(),
});
 
envVariables.parse(process.env);
 
declare global {
  namespace NodeJS {
    interface ProcessEnv extends z.infer<typeof envVariables> {}
  }
}
```

File: backend/controllers/subscription.ts (hunk 1 above)
Comment (kiritocode1) [reply backend/controllers/subscription.ts:8]: this is a bad idea , someone could use this to  inject sql , please use a javascript tempelate literal function to validate the name param before running this

File: backend/config/index.ts (hunk 4 above)
Comment (kiritocode1) [reply backend/config/index.ts:28]: this allows you to extend process.env  with proper typesafety in check

File: backend/routes/index.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        message: "Welcome to our API homepage!",
+      });
+    } catch (err) {
+      res.status(500).json({
+        status: "failed",
+        data: [],
+        message: "Internal Server Error",
+      });
+    }
+  });
```
Comment (Codimow) [inline backend/routes/index.ts:20]: # Better code with better error handling

```
const Router = (server: Application) => {
  // home route with the get method and a handler

  server.get("/home", (req: Request, res: Response) => {
    res.status(200).json({
      status: "success",
      data: [],
      message: "Welcome to our API homepage!",
    });
  });

  // improved error handling
  server.use((err: Error, req: Request, res: Response, next: Function) => {
    console.error(err.stack);
    res.status(500).json({
      status: "failed",
      data: [],
      message: "Internal Server Error",
    });
  });
```

**Review by Codimow — COMMENTED** [review]
js better code sarvesh bhai u should learn from me (:, js kidding

File: backend/utils/postgres.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  user: DB_USER,
+  host: DB_HOST,
+  database: DB_DATABASE,
+  password: DB_PASSWORD,
+  port: parseInt(DB_PORT || "5432"),
+  ssl: SSL === "true" ? true : SSL === "false" ? false : undefined,
+});
+
+// Function to run a query
+export async function runQuery(queryText: string, values: any[] = []): Promise<any> {
```
Comment (kiritocode1) [inline backend/utils/postgres.ts:15]: why is type sign here returning Promise<any> , if we know the explisit types could be Promise<Rows| Error> 

a good explaination could work to favour this change

File: backend/routes/index.ts (hunk 5 above)
Comment (kiritocode1) [reply backend/routes/index.ts:20]: why tho , next() function also exists on the get / i feel like mentioning this is just not good for performance as this router would have to invoke another function just to gain these parameters

File: backend/utils/postgres.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  try {
+    // Use a connection from the pool to execute the query
+    const res = await pool.query(queryText, values);
+    return res.rows;
+  } catch (error) {
+    console.error("Database query error:", error);
+    throw new Error(`Database query failed: ${error}`);
```
Comment (Codimow) [inline backend/utils/postgres.ts:22]: this is better for debugging

```
throw new Error(`Database query failed: ${error.message}`);
```

File: backend/config/index.ts (hunk 4 above)
Comment (sarvesh371) [reply backend/config/index.ts:28]: Updated

File: backend/utils/postgres.ts (hunk 7 above)
Comment (sarvesh371) [reply backend/utils/postgres.ts:22]: Done

**Pull request merged**